cpp/src/aztec/stdlib/encryption/schnorr/schnorr.cpp

```diff
@@ -293,8 +293,9 @@ void verify_signature(const byte_array<C>& message, const point<C>& pub_key, con
 
     field_t<C> output_hi(output.slice(0, 16));
     field_t<C> output_lo(output.slice(16, 16));
-    output_lo.assert_equal(sig.e_lo, "verify signature failed");
```

Comment:
One thing to note is that if the current behaviour has a minor issue:

It is not currently possible to distinguish between the signature being invalid, and a Prover deliberately providing incorrect witness values.  

This isn't a problem for the Aztec Connect use case, but it may be an issue for a Noir app where proof computation is being delegated to a third party (i.e. you can't distinguish between "I can't make a valid proof because you gave me a bad signature" and "I *won't* make a valid proof because I'm dishonest"

Reply:
I've re-added the `assert_equal` calls, which I mistakenly removed when I made the change to the logic to add actual constraints (I was confused by the "equal" name and "==" (equal operator) and thought they were the same thing).

I'm pretty sure Noir just needs to be able to know the that something was valid/invalid via return type (and if I understand correctly, that needs to be constrained)

Reply:
@zac-williamson so perhaps remove the `assert_equal`s and rename to `verify_signature_unsafe` or `prove_signature_verification_result` or something like this?